test/integration/targets/order/order.yml

```diff
@@ -23,5 +23,17 @@
           - item.1 == shostlist[item.0]
       loop: '{{ lookup("indexed_items", inputlist) }}'
       vars:
-        hostlist: '{{lookup("file", "hostlist.txt").splitlines()}}'
-        shostlist: '{{lookup("file", "shostlist.txt").splitlines()}}'
+        hostlist: '{{ lookup("file", "hostlist.txt").splitlines() }}'
+        shostlist: '{{ lookup("file", "shostlist.txt").splitlines() }}'
+      when: myorder | default('inventory') != 'shuffle'
+
+    - name: Assert that shuffle worked
+      assert:
+        that:
+          - item.1 != hostlist[item.0] or item.1 in hostlist
```

Comment:
I realize this isn't a true test for randomness. The `or` statement is necessary in the cases where the test string happens to be the first item in the list.